CityPuzzle/Main Game/ComplitedPage.xaml.cs

```diff
@@ -29,5 +29,17 @@ private void NewPuzzle_clicked(object sender, EventArgs e)
         {
             Navigation.PushAsync(new QuestPage(questsList));
         }
+        protected override void OnDisappearing()// GAL reiktu tam klase nauja sukurt (Navigation)
```

Comment:
I believe that this should be ineed in a seperate class in a static method that is used consistantly over the whole project to clear the stack when necessary